.github/workflows/ci.yaml

```diff
@@ -0,0 +1,155 @@
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Install build dependencies needed for VPU plugin
+        run: |
+          sudo apt-get update
+          sudo apt-get -y install libusb-1.0-0-dev
+      - name: Install tools
+        run: |
+          # checkers
+          cd /
+          go get -v golang.org/x/lint/golint
+          go get -v github.com/fzipp/gocyclo
+          go get -v golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow
+      - run: make go-mod-tidy
+      - run: make format
+      - run: make lint
+      - run: make BUILDTAGS=kerneldrv
+      - run: make vet
+      - run: make cyclomatic-check
+      - run: make test BUILDTAGS=kerneldrv
+      - name: Make vendor folder
+        run: make -e vendor
+      - uses: actions/upload-artifact@v1
+        with:
+          name: vendor
+          path: vendor
+
+  image:
+    name: Build image
+    needs: [tools, build]
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        image:
+          - intel-fpga-admissionwebhook
+          - intel-fpga-initcontainer
+          - intel-gpu-plugin
+          - intel-fpga-plugin
+          - intel-qat-plugin
+          - intel-vpu-plugin
+
+          # Demo images
+          - crypto-perf
+          - opae-nlb-demo
+          - openssl-qat-engine
+          - ubuntu-demo-opencl
+          - ubuntu-demo-openvino
+        builder: [buildah, docker]
+    steps:
+      - uses: actions/setup-go@v2-beta
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Set up environment
+        if: matrix.builder == 'buildah'
+        shell: bash
+        run: |
+          echo "::set-env name=GO_PATH::$(go env GOPATH)"
+      - uses: actions/cache@v1
+        id: cache-buildah
+        name: Fetch cached buildah
+        if: matrix.builder == 'buildah'
+        with:
+          path: ${{ env.GO_PATH }}
+          key: ${{ runner.os }}-buildah-${{ env.BUILDAH_VERSION }}
+      - name: Install buildah
+        if: matrix.builder == 'buildah'
+        run: |
+          sudo apt-get update
+          sudo apt-get -y install libgpgme11 libprotobuf10
+          cd ${GO_PATH}/src/github.com/containers/buildah
+          sudo cp buildah /usr/local/bin
+          # configure buildah
+          sudo mkdir -p /etc/containers
+          sudo mkdir -p /etc/cni/net.d
+          sudo mkdir -p /opt/cni/bin
+          sed -i -e 's/build.sh/build_linux.sh/' Makefile
+          make install.cni.sudo
+          echo -e '[registries.search]\nregistries = ["docker.io"]\n\n' | sudo tee /etc/containers/registries.conf
+          sudo curl https://raw.githubusercontent.com/kubernetes-sigs/cri-o/${CRIO_VERSION}/test/policy.json -o /etc/containers/policy.json
+          # install runc
+          sudo curl -L https://github.com/opencontainers/runc/releases/download/${RUNC_VERSION}/runc.amd64 -o /usr/bin/runc
+          sudo chmod +x /usr/bin/runc
+      - uses: actions/checkout@v2
```

Comment:
do we need to checkout here?

Reply:
Yes, this way we test changes in the `Dockerfile`s used for image building.

Reply:
OK. If `build` checks out the code and `image` depends it, I thought this is redundant

Reply:
No, every job runs in its own VM.